### PR TITLE
FFM-9213: Implement targets for the feature flag terraform resource

### PR DESCRIPTION
**Pull request by ribeirophillipe**
## Describe your changes
* Add targets rule for feature flags
* update instructions definition
* include mapping for the instruction types

## Comment Triggers

<details>
  <summary>PR Check triggers</summary>
  
- Build: `trigger build`
- Sub Category Field Check: `trigger subcategoryfieldcheck`
- gitleaks: `trigger gitleaks`

### CODE REVIEW

File: examples/resources/harness_platform_feature_flag/resource.tf [hunk 1]
```diff
@@ -60,4 +60,111 @@ resource "harness_platform_feature_flag" "mymultivariateflag" {
...
+    description = "Free trial period 20 days"
+    value       = "20"
+  }
+
+  instructions {
+    kind = "addRule"
+    parameters = {
+      serve = {
+        distribution = {
+          bucketBy = "identifier",
```
Comment (davejohnston) [inline examples/resources/harness_platform_feature_flag/resource.tf:147]: I don't think we want to expose this to the customer just yet.  They can't change this via the UI either.  We should just default to it on the backend

File: examples/resources/harness_platform_feature_flag/resource.tf [hunk 2]
```diff
@@ -60,4 +60,111 @@ resource "harness_platform_feature_flag" "mymultivariateflag" {
...
+            },
+            {
+                variation = "name"
+                weight = 40
+            }
+          ]
+        }
+      }
+      clauses = {
+        op =  "segmentMatch",
```
Comment (davejohnston) [inline examples/resources/harness_platform_feature_flag/resource.tf:165]: I don't think we should expose this operator - it doesn't really make sense to the customer to have to learn our internal operators.  From there point of view, its a target group rule

File: docs/resources/platform_feature_flag.md [hunk 3]
```diff
@@ -76,6 +76,113 @@ resource "harness_platform_feature_flag" "mymultivariateflag" {
...
+  }
+
+  variation {
+    identifier  = "trial20"
+    name        = "20 days trial"
+    description = "Free trial period 20 days"
+    value       = "20"
+  }
+
+  instructions {
```
Comment (davejohnston) [inline docs/resources/platform_feature_flag.md:114]: This looks functional, but I don't think we should try and simplify this for the user.  For example could this just be written as 
```
addTargetRules {
         variation = "enabled"
         targets = ["targets1", "targets2"]
}
```
and on the backend we convert it into the internal payload that the ff server requires.

File: docs/resources/platform_feature_flag.md [hunk 4]
```diff
@@ -76,6 +76,113 @@ resource "harness_platform_feature_flag" "mymultivariateflag" {
...
```
Comment (davejohnston) [inline docs/resources/platform_feature_flag.md:157]: A bit like the target rule, could we simplify this for the user?  The only bits the care about are:
the group name and the variation.

If its a % rollout rule then they care about the distribution as well.

File: examples/resources/harness_platform_feature_flag/resource.tf (hunk 1 above)
Comment (ribeirophillipe) [reply examples/resources/harness_platform_feature_flag/resource.tf:147]: ok

File: docs/resources/platform_feature_flag.md (hunk 3 above)
Comment (ribeirophillipe) [reply docs/resources/platform_feature_flag.md:114]: I agree... updated this model

File: docs/resources/platform_feature_flag.md (hunk 4 above)
Comment (ribeirophillipe) [reply docs/resources/platform_feature_flag.md:157]: I've modified this.

File: examples/resources/harness_platform_feature_flag/resource.tf (hunk 2 above)
Comment (ribeirophillipe) [reply examples/resources/harness_platform_feature_flag/resource.tf:165]: Agreed

File: internal/service/platform/feature_flag/resource_feature_flag.go [hunk 5]
```diff
@@ -157,6 +284,45 @@ type FFOpts struct {
 	Variations          []nextgen.Variation `json:"variations"`
 }
 
+// FFPatchOpts is the options for patching a feature flag
+type FFPatchOpts struct {
+	Identifier          string              `json:"identifier"`
```
Comment (rathodmeetsatish) [inline internal/service/platform/feature_flag/resource_feature_flag.go:289]: usually we keep these objects in harness-go-sdk that way resource file will be cleaner @davejohnston @ribeirophillipe

File: docs/resources/platform_feature_flag.md [hunk 6]
```diff
@@ -76,6 +76,103 @@ resource "harness_platform_feature_flag" "mymultivariateflag" {
...
+  variation {
+    identifier  = "trial20"
+    name        = "20 days trial"
+    description = "Free trial period 20 days"
+    value       = "20"
+  }
+
+  add_target_groups_rules {
+    group_name = "group_name"
+    variation = "enabled"
```
Comment (davejohnston) [inline docs/resources/platform_feature_flag.md:156]: Same here, we should use a variation that has been defined as part of the flag.

File: internal/service/platform/feature_flag/resource_feature_flag.go [hunk 7]
```diff
@@ -157,6 +284,45 @@ type FFOpts struct {
...
+		return nil
+	}
+
+	qp := buildFFQueryParameters(d)
+	opts := buildFFPatchOpts(d)
+
+	feature, httpResp, err := c.FeatureFlagsApi.PatchFeature(ctx, c.AccountId, qp.OrganizationId, qp.ProjectId, id, opts)
+
+	if err != nil {
+		return helpers.HandleApiError(err, d, httpResp)
```
[the remark this answers is not in the document]
Comment (ribeirophillipe) [reply internal/service/platform/feature_flag/resource_feature_flag.go:318]: yes!

**Review by rathodmeetsatish — APPROVED** [review]

**Review by davejohnston — APPROVED** [review]

**Pull request merged**